### PR TITLE
buf: 1.31.0 -> 1.32.2

### PR DESCRIPTION
**Pull request by aaronjheng**
## Description of changes

https://github.com/bufbuild/buf/releases/tag/v1.32.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by gepbird — CHANGES_REQUESTED** [review]

**Review by gepbird — CHANGES_REQUESTED** [review]
LGTM! Just one more thing, please remove this outdated comment in the patches: `# and a test which requires the source checkout to be part of a git repository`

**Comment by gepbird** [comment]
Result of `nixpkgs-review pr 313092` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>buf</li>
    <li>dart-sass</li>
    <li>dart-sass.pubcache</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>ignite-cli</li>
    <li>jekyll</li>
    <li>jsonnet</li>
    <li>jsonnet.doc</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>shopware-cli</li>
  </ul>
</details>

**Comment by gepbird** [comment]
How did you get that hash? I get a mismatch when running nixpkgs-review on this PR:
```
error: hash mismatch in fixed-output derivation '/nix/store/nhn7sz52wqr1ghrki1mdf0clddgh2pkv-source.drv':                                                                                                         
         specified: sha256-ONeYEfiEy7I2uHZ51n1gdm4iAfz/z8ZvPUfvmX7dd34=                                                                                                                                           
            got:    sha256-BnyFx0FphRmqBRvmPrFScU4/lOjQHO79oeO1JysJEE8= 
```

**Comment by aaronjheng** [comment]
That's odd. Fixing.

**Review by gepbird — APPROVED** [review]

File: pkgs/development/tools/buf/default.nix [hunk 1]
```diff
@@ -10,16 +10,16 @@
...
   pname = "buf";
-  version = "1.31.0";
+  version = "1.32.1";
 
   src = fetchFromGitHub {
     owner = "bufbuild";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-CNt7ZfUMRQNNVusaBgEW64WH1TFYz8yEBfKksGVRVUM=";
+    hash = "sha256-VTScGJBcrrAG5p3irMsTrW13kstiNXR/oDQ2NVX2hiI=";
```
[the remark this answers is not in the document]
Comment (aaronjheng) [reply pkgs/development/tools/buf/default.nix]: Failed partially again. Let's leave it as-is for now.

[the remark this answers is not in the document]
Comment (gepbird) [reply pkgs/development/tools/buf/default.nix]: @aaronjhang would you mind trying this again?

The order is important, make sure to obtain the hash after you made version and patch changes:
1. remove the patch that disables `TestWorkspaceGit`
2. in fetchFromGitHub add `leaveDotGit = true`, and fake the hash: `hash= ""`
3. try to build the package with `nix build .#buf` to obtain the hash, then replace the fake hash with it
4. build the package again to make sure it compiles

[the remark this answers is not in the document]
Comment (aaronjheng) [reply pkgs/development/tools/buf/default.nix]: Can do.

**Comment by gepbird** [comment]
Another hash mismatch, weird... I assume you consistently get the same hash, maybe you have some uncommitted changes that causes the hash to be different?

```
error: hash mismatch in fixed-output derivation '/nix/store/z6r6nxvvwgg31vj75gs5hardnbp4zlfp-source.drv':
         specified: sha256-WCMQr9IH3n3T1fVm3rf5zeCMsdYDMuJQVF+83MyTP6k=
            got:    sha256-fX4n6WpkbASu+wFEp4hr3usOCbfKSlZvrEywsN7Xtd0=
error: 1 dependencies of derivation '/nix/store/lp89a8ikra845hvn7r9dcm16lbv13mqq-buf-1.32.2.drv' failed to build
```

@wegank do you have any idea why do we get different hashes?

**Comment by wegank** [comment]
Because `leaveDotGit = true;` is used without a `postFetch` that does some cleaning. An example is the following:

https://github.com/NixOS/nixpkgs/blob/f478e1cb40ff17ef02ab22f45e0512868b0c86c6/pkgs/by-name/ll/llama-cpp/package.nix#L82-L87

**Comment by gepbird** [comment]
Unfortunately this just deletes the .git directory and the test is failing again:
```
buf> --- FAIL: TestWorkspaceGit (0.02s)
<some lines are omitted>
buf>            Messages:       expected exit code 1 to be one of [0]
buf>                            :args: 'build' '../../../../.git#ref=HEAD,subdir=private/buf/cmd/buf/testdata/workspace/success/dir/proto'
buf>                            stdout: 
buf>                            stderr: Failure: could not clone file:///build/source/.git: exit status 128
buf>                            fatal: '/build/source/.git' does not appear to be a git repository
buf>                            fatal: Could not read from remote repository.
buf>                        
buf>                            Please make sure you have the correct access rights
buf>                            and the repository exists.
```

**Comment by 06kellyjac** [comment]
Yes, please continue to skip that test and avoid `leaveDotGit`. Thanks

Also the patch can probably be dropped now in favor of `checkFlags = [ "--skip=Something|SomethingElse" ];`

**Review by gepbird — APPROVED** [review]
Result of `nixpkgs-review pr 313092` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>buf</li>
    <li>dart-sass</li>
    <li>dart-sass.pubcache</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>ignite-cli</li>
    <li>jekyll</li>
    <li>jsonnet</li>
    <li>jsonnet.doc</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>shopware-cli</li>
  </ul>
</details>

**Pull request merged**